crates/tako/src/scheduler/state.rs

```diff
@@ -83,8 +92,20 @@ impl SchedulerState {
         task: &Task,
         taskmap: &TaskMap,
         worker_map: &Map<WorkerId, Worker>,
+        try_stick: bool,
     ) -> Option<WorkerId> {
-        self.tmp_workers.clear();
+
+        // Fast path
+        if try_stick && task.inputs.is_empty() {
+            if let Some(worker_id) = self.get_last() {
+                if worker_map[&worker_id].have_immediate_resources_for_rq(&task.configuration.resources) {
```

Comment:
Why does this use `have_immediate_resources_for_rq`, but below `is_capable_to_run` is used? The new task might be different from the previous one, right?

Reply:
Added comment into the code